contracts/vaults/RibbonDeltaVault.sol

```diff
@@ -146,7 +146,36 @@ contract RibbonDeltaVault is RibbonVault, OptionsDeltaVaultStorage {
      * @notice Closes the existing long.
      *         This allows all the users to withdraw if the next option is malicious.
      */
-    function commitAndClose() external onlyOwner nonReentrant {}
+    function commitAndClose() external onlyOwner nonReentrant {
+        address oldOption = optionState.currentOption;
+
+        optionState.nextOption = counterpartyThetaVault
+            .optionState()
+            .nextOption;
+        optionState.nextOptionReadyAt = uint32(block.timestamp.add(delay));
+
+        _closeLong(oldOption);
+    }
+
+    /**
+     * @notice Closes the existing long position for the vault.
+     */
+    function _closeLong(address oldOption) private {
+        optionState.currentOption = address(0);
+        vaultState.lastLockedAmount = vaultState.lockedAmount;
+        vaultState.lockedAmount = 0;
```

Comment:
Does the `lockedAmount` variable even make sense in the context of Delta vault? We're not locking anything and spending it right away, so we *could* be double counting here

Reply:
it doesn't _really_ make sense, but trying to abstract as much as possible. I can repurpose locked amount to be the premium paid during the week. used in RibbonVault parent in totalBalance() 

if you think its an unnecessary stretch can just keep locked lingo specific to theta vault and introduce something new

Reply:
Gotcha, thanks for the explanation